lms/djangoapps/teams/teams_features.py

```diff
@@ -1,124 +1,41 @@
-        """
-        if self.section:
-            subsections = self.section.get_children()
-            if len(subsections)==1:
-                return subsections[0]
-            else:
-                LOG.warning("Only one subsection must be defined")
-        return None
-
-    def _find_unit(self):
-        """
-        Finds the rocketchat's unit.
-        """
-        if self.subsection:
-            units = self.subsection.get_children()
-            if len(units)==1:
-                return units[0]
-            else:
-                LOG.warning("Only one unit must be defined")
-        return None
-
-    def _prefetch_and_bind_course(self, request):
-        """
-        Prefetches all descendant data for the requested section and
-        sets up the runtime, which binds the request user to the section.
-        """
-        self.field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
-            self.course_key,
-            self.effective_user,
-            self.course,
-            depth=CONTENT_DEPTH,
-            read_only=CrawlersConfig.is_crawler(request),
-        )
-        self.course = get_module_for_descriptor(
-            self.effective_user,
-            request,
-            self.course,
-            self.field_data_cache,
-            self.course_key,
-            course=self.course,
-        )
-
-    def discussion_is_available(self):
-        """
-        This method checks if teams are using discussion
-        """
-        if "discussion_is_available" in self.course.teams_configuration:
-            return self.course.teams_configuration["discussion_is_available"]
-        return True
+        self.teams_configuration = course.teams_configuration
 
-    def get_rocket_chat_locator(self):
+    def get_replacement_locator(self):
         """
-        This method gets the locator for a unique component in a unit.
+        This method returns the locator for a unit, to do that this gets the unit location_id
+        from teams_configuration, to set that go to studion->advanced settings-> teams_configuration
         """
-        if "rocketchat-xblock-id" in self.course.teams_configuration:
-            block_id = self.course.teams_configuration["rocketchat-xblock-id"]
+        try:
+            block_id = self.teams_configuration["replacement_location_id"]
             block_type = "vertical"
             locator = BlockUsageLocator(self.course_key, block_type, block_id)
-        else:
-            unit = self._find_unit()
-            child = unit.get_children()
-            child = child[0]
-            locator = child.parent
-        return locator.to_deprecated_string()
-
-def get_users_enrolled(course_key):
-    """Return a generator with the username of every user in the course"""
-    users = CourseEnrollmentManager().users_enrolled_in(course_key)
-    for user in users:
-        yield user.username
+            return locator
+        except (KeyError, InvalidKeyError):
```

Comment:
InvalidKeyError exception is raised when BlockUsageLocator functions fails?

Reply:
when a BlockUsageLocator has a  block_id  None and the deprecated status of the course key is false